cpp/src/arrow/testing/random.h

```diff
@@ -433,12 +433,18 @@ class ARROW_TESTING_EXPORT RandomArrayGenerator {
   ///
   /// \param[in] size the size of the array to generate
   /// \param[in] byte_width the byte width of fixed-size binary items
+  /// \param[in] min_byte the lower bound of each byte in the binary determined by the
+  ///            uniform distribution
+  /// \param[in] max_byte the upper bound of each byte in the binary determined by the
+  ///            uniform distribution
   /// \param[in] null_probability the probability of a value being null
   /// \param[in] alignment alignment for memory allocations (in bytes)
   /// \param[in] memory_pool memory pool to allocate memory from
   ///
   /// \return a generated Array
   std::shared_ptr<Array> FixedSizeBinary(int64_t size, int32_t byte_width,
+                                         uint8_t min_byte = static_cast<uint8_t>('A'),
+                                         uint8_t max_byte = static_cast<uint8_t>('z'),
```

Comment:
Is it possible to still expose the old signature as well for convenience? In many cases there's no need to override `min_byte` and `max_byte`, while still wanting a non-zero null probability.

Reply:
Sounds good, will do!

Reply:
Instead of adding a new overload, I put the `min_byte` and `max_byte` after the current `null_probability` so the caller is still able to specify a non-zero `null_probability` w/o having to override `min_byte` and `max_byte`.

How do you think?

Reply:
Ah, that sounds good, thank you.